### ISSUE TITLE
MD0029 triggers if code block is part of unordered list

### ISSUE THREAD

**Issue by srenfo** [markdownlint/markdownlint#299]
## Description
Rule MD0029 is triggered inside code blocks that are part of unordered lists.

### Environment

**MDL Version** 0.9.0

### Expected Behavior

The rule should not be triggered.
### Actual Behavior

The rule is triggered.

## Replication Case

    - foobar
      ```
      1. foo
      2. bar
      ```
```sh
$ mdl foobar.md 
foobar.md:4: MD029 Ordered list item prefix
foobar.md:2: MD031 Fenced code blocks should be surrounded by blank lines

A detailed description of the rules is available at https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md
```
If you dedent the code block or insert blank lines around the block (as per MD031) then the rule will not be triggered - presumably because it is now separate from the list item. But I would expect different rendering results between those cases and the replication case above.

**Comment by srenfo** [comment]
I have updated this to version 0.9.0 and expanded and fixed the replication case.

**Comment by srenfo** [comment]
To clarify why we don't want a blank line between list item and code block:

### no blank line, code block indented
~~~
- foo
- bar
  ```
  code
  ```
~~~

- foo
- bar
  ```
  code
  ```

### blank line, code block indented

~~~
- foo
- bar

  ```
  code
  ```
~~~

- foo
- bar

  ```
  code
  ```

### blank line, code block not indented
~~~
- foo
- bar

```
code
```
~~~

- foo
- bar

```
code
```

Notice how the blank line between code block and list item is rendered in the latter two cases. Also, in the middle example there's somehow another blank line between the list items. GitLab renders this the same way.

**Comment by marcofranssen** [comment]
Similar happens when using ordered lists #323 

It also struggles with quotations.

**Comment by jaymzh** [comment]
This is an issue with the Kramdown parser - it doesn't recognize the indentation at all here, which sorta ties our hands...

**Comment by jaymzh** [comment]
Actually I can't repro this at all:

Markdown:
    # foo
    
    * foo
      ```
      a code block
      ```
    * bar
    
    Some more stuff.

Result:
```
/tmp/test.md:4: MD031 Fenced code blocks should be surrounded by blank lines
/tmp/test.md:6: MD031 Fenced code blocks should be surrounded by blank line
```

And if I do that:

    # foo
    
    * foo
    
      ```
      a code block
      ```

    * bar
    
    Some more stuff.

Then I get no errors.

So this appears fixed on main branch.
